File: Python/Program/Vehicle.py
class Vehicle:
    def __init__(self, plat="", merk="", warna="", tahun=0):
        self.plat = plat
        self.merk = merk
        self.warna = warna
        self.tahun = tahun

    def getPlat(self):
        return self.plat

    def getMerk(self):
        return self.merk

    def getWarna(self):
        return self.warna

    def getTahun(self):
        return self.tahun

    def setPlat(self, plat):
        self.plat = plat

    def setMerk(self, merk):
        self.merk = merk

    def setWarna(self, warna):
        self.warna = warna

    def setTahun(self, tahun):
        self.tahun = tahun
    
    def ketemu(self,llist, platt): # mencari id anggota
        for kendaraan in llist:
            if kendaraan.getPlat() == platt:
                return 1
        return 0

File: C++/Program/Vehicle.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Vehicle {
	private:
		string plat;
		string merk;
		string warna;
		int tahun;
		string tipe;
		
	public:
		// konstrukt secara langsung di parameter
		Vehicle(string plat = "", string merk = "", string warna = "", int tahun = 0, string tipe = "") : plat(plat), merk(merk), warna(warna), tahun(tahun), tipe(tipe) 
		{
			
		}
		// getter
		string getPlat() {
			return this->plat;
		}

		string getMerk() {
			return this->merk;
		}

		string getWarna() {
			return this->warna;
		}

		int getTahun() {
			return this->tahun;
		}

		string getType() {
			return this->tipe;
		}
		// setter
		void setPlat(string plat) {
			this->plat = plat;
		}

		void setMerk(string merk) {
			this->merk = merk;
		}

		void setWarna(string warna) {
			this->warna = warna;
		}

		void setTahun(int tahun) {
			this->tahun = tahun;
		}

		void setType(string tipe) {
			this->tipe = tipe;
		}
		
		// mencari plat yang sama
		bool ketemu(vector<Vehicle*> llist, string platt) 
		{
			for (Vehicle* kendaraan : llist) 
			{
				if (kendaraan->getPlat() == platt) 
				{
					return true;
				}
			}
			return false;
		}
		
		virtual ~Vehicle() // virtual ini agar dynamic_cast nya berjalan lancar di car dan motorcycle
		{
			
		}
};
